### PR TITLE
[NO-ISSUE] Added docker setup

### PR DESCRIPTION
**Pull request by Epitrochoid**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,9 +4,13 @@
   "description": "Watches package repo RSS feeds (e.g. PyPi and Maven) and pushes incoming changes to a RabbitMQ event queue",
   "main": "src/server.js",
   "scripts": {
+    "docker:build": "docker-compose build",
+    "docker:start": "docker-compose up rediscluster rabbitmq",
+    "docker:stop": "docker-compose down",
     "lint": "eslint .",
     "start": "node src/server.js",
-    "test": "ava"
+    "test": "ava test",
```
Comment (ctdio) [inline package.json]: You should having this command be `docker-compose run test; npm run docker:stop`. That way we default to running tests in docker.

**Review by ctdio — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.dockerignore
```
Comment (Druotic) [inline .gitignore]: Why ignore .dockerignore? Usually you'd want to check this in so that everyone has consistent behavior.

File: package.json [hunk 3]
```diff
@@ -40,6 +44,7 @@
     "feedparser": "^2.2.1",
     "request": "^2.81.0",
     "request-promise": "^4.2.1",
+    "require-self-ref": "^2.0.1",
```
Comment (Druotic) [inline package.json:47]: Just realized you're using this module already but just now adding it to package.json. An automated test job would have caught this. We should get travis set up to run jobs automatically on push/pr, like the other repos.

**Review by Druotic — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+node_modules
+package-lock.json
```
Comment (Druotic) [inline .gitignore]: You should also check in `package-lock.json`.

File: package.json (hunk 3 above)
Comment (Epitrochoid) [reply package.json:47]: I'll make a separate ticket for this.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
+FROM node:8.1.2-alpine
+
+ENV HOME=/usr/rss-watcher
+
+RUN mkdir -p $HOME
+
+WORKDIR $HOME
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash git openssh
```
Comment (ctdio) [inline Dockerfile:10]: Just want to note that this is pretty important for other repos that adopt a similar setup. Without this, we can't pull in the `service-util` code.

File: package.json [hunk 6]
```diff
@@ -4,9 +4,14 @@
...
   "scripts": {
+    "docker:build": "docker-compose build",
+    "docker:start": "docker-compose up rediscluster rabbitmq",
+    "docker:stop": "docker-compose down",
     "lint": "eslint .",
     "start": "node src/server.js",
-    "test": "ava"
+    "test": "docker-compose run test; npm run docker:stop",
+    "test:docker:ci": "ava test",
+    "path": "which docker-compose"
```
Comment (ctdio) [inline package.json]: you can probably drop this command

**Review by ctdio — APPROVED** [review]

**Review by austinkelleher — APPROVED** [review]

**Pull request merged**